### ISSUE TITLE
Issue compiling with Teensyduino 1.58 on PIO

### ISSUE THREAD

**Issue by bimac** [luni64/TeensyStep#151]
Hi luni64,

As mentioned on the [PJRC Forum](https://forum.pjrc.com/threads/72534-PlatformIO-update-for-1-58-please-help-test?p=323785&viewfull=1#post323785) I can't seem to compile TeensyStep with Teensyduino 1.58 on PlatformIO:

**`platformio.ini`:**
```ini
[env:teensy35]
platform = teensy
framework = arduino
board = teensy35
lib_deps = https://github.com/luni64/TeensyStep.git
```

**GCC output:**
<details>
<summary>click to expand</summary>

```
In file included from /home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/WProgram.h:46,
                 from /home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/Arduino.h:6,
                 from .pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:1:
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp: In member function 'TeensyStep::Stepper& TeensyStep::Stepper::setPullInSpeed(int32_t)':
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:141:16: error: expected unqualified-id before '(' token
  141 | #define abs(x) ({ \
      |                ^
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:90:35: note: in expansion of macro 'abs'
   90 |         vPullIn = vPullOut = std::abs(speed);
      |                                   ^~~
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:142:3: error: expected primary-expression before 'typeof'
  142 |   typeof(x) _x = (x); \
      |   ^~~~~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:90:35: note: in expansion of macro 'abs'
   90 |         vPullIn = vPullOut = std::abs(speed);
      |                                   ^~~
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:142:3: error: expected '}' before 'typeof'
  142 |   typeof(x) _x = (x); \
      |   ^~~~~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:90:35: note: in expansion of macro 'abs'
   90 |         vPullIn = vPullOut = std::abs(speed);
      |                                   ^~~
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:141:17: note: to match this '{'
  141 | #define abs(x) ({ \
      |                 ^
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:90:35: note: in expansion of macro 'abs'
   90 |         vPullIn = vPullOut = std::abs(speed);
      |                                   ^~~
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:142:3: error: expected ')' before 'typeof'
  142 |   typeof(x) _x = (x); \
      |   ^~~~~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:90:35: note: in expansion of macro 'abs'
   90 |         vPullIn = vPullOut = std::abs(speed);
      |                                   ^~~
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:141:16: note: to match this '('
  141 | #define abs(x) ({ \
      |                ^
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:90:35: note: in expansion of macro 'abs'
   90 |         vPullIn = vPullOut = std::abs(speed);
      |                                   ^~~
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:143:4: error: '_x' was not declared in this scope
  143 |   (_x > 0) ? _x : -_x; \
      |    ^~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:90:35: note: in expansion of macro 'abs'
   90 |         vPullIn = vPullOut = std::abs(speed);
      |                                   ^~~
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:144:1: warning: no return statement in function returning non-void [-Wreturn-type]
  144 | })
      | ^
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:90:35: note: in expansion of macro 'abs'
   90 |         vPullIn = vPullOut = std::abs(speed);
      |                                   ^~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp: At global scope:
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:144:2: error: expected unqualified-id before ')' token
  144 | })
      |  ^
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:90:35: note: in expansion of macro 'abs'
   90 |         vPullIn = vPullOut = std::abs(speed);
      |                                   ^~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:91:9: error: expected unqualified-id before 'return'
   91 |         return *this;
      |         ^~~~~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:94:5: error: 'Stepper' does not name a type
   94 |     Stepper& Stepper::setPullInOutSpeed(int32_t pullInSpeed, int32_t pullOutSpeed)
      |     ^~~~~~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:102:10: error: 'Stepper' has not been declared
  102 |     void Stepper::setTargetAbs(int32_t target)
      |          ^~~~~~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp: In function 'void setTargetAbs(int32_t)':
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:104:31: error: 'current' was not declared in this scope
  104 |         setTargetRel(target - current);
      |                               ^~~~~~~
Compiling .pio/build/teensy35/FrameworkArduino/HardwareSerial.cpp.o
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:104:9: error: 'setTargetRel' was not declared in this scope; did you mean 'setTargetAbs'?
  104 |         setTargetRel(target - current);
      |         ^~~~~~~~~~~~
      |         setTargetAbs
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp: At global scope:
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:107:10: error: 'Stepper' has not been declared
  107 |     void Stepper::setTargetRel(int32_t delta)
      |          ^~~~~~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp: In function 'void setTargetRel(int32_t)':
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:109:9: error: 'setDir' was not declared in this scope
Compiling .pio/build/teensy35/FrameworkArduino/HardwareSerial1.cpp.o
  109 |         setDir(delta < 0 ? -1 : 1);
      |         ^~~~~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:110:9: error: 'target' was not declared in this scope
  110 |         target = current + delta;
      |         ^~~~~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:110:18: error: 'current' was not declared in this scope
  110 |         target = current + delta;
      |                  ^~~~~~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:111:9: error: 'A' was not declared in this scope; did you mean 'A9'?
  111 |         A = std::abs(delta);
      |         ^
      |         A9
In file included from /home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/WProgram.h:46,
                 from /home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/Arduino.h:6,
                 from .pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:1:
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:141:16: error: expected unqualified-id before '(' token
  141 | #define abs(x) ({ \
      |                ^
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:111:18: note: in expansion of macro 'abs'
  111 |         A = std::abs(delta);
      |                  ^~~
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:142:3: error: expected primary-expression before 'typeof'
  142 |   typeof(x) _x = (x); \
      |   ^~~~~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:111:18: note: in expansion of macro 'abs'
  111 |         A = std::abs(delta);
      |                  ^~~
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:142:3: error: expected '}' before 'typeof'
  142 |   typeof(x) _x = (x); \
      |   ^~~~~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:111:18: note: in expansion of macro 'abs'
  111 |         A = std::abs(delta);
      |                  ^~~
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:141:17: note: to match this '{'
  141 | #define abs(x) ({ \
      |                 ^
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:111:18: note: in expansion of macro 'abs'
  111 |         A = std::abs(delta);
      |                  ^~~
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:142:3: error: expected ')' before 'typeof'
  142 |   typeof(x) _x = (x); \
      |   ^~~~~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:111:18: note: in expansion of macro 'abs'
  111 |         A = std::abs(delta);
      |                  ^~~
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:141:16: note: to match this '('
  141 | #define abs(x) ({ \
      |                ^
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:111:18: note: in expansion of macro 'abs'
  111 |         A = std::abs(delta);
      |                  ^~~
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:143:4: error: '_x' was not declared in this scope
  143 |   (_x > 0) ? _x : -_x; \
      |    ^~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:111:18: note: in expansion of macro 'abs'
  111 |         A = std::abs(delta);
      |                  ^~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp: At global scope:
/home/bimac/.platformio/packages/framework-arduinoteensy/cores/teensy3/wiring.h:144:2: error: expected unqualified-id before ')' token
Compiling .pio/build/teensy35/FrameworkArduino/HardwareSerial2.cpp.o
  144 | })
      |  ^
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:111:18: note: in expansion of macro 'abs'
  111 |         A = std::abs(delta);
      |                  ^~~
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:112:5: error: expected declaration before '}' token
  112 |     }
      |     ^
.pio/libdeps/teensy35/TeensyStep/src/Stepper.cpp:113:1: error: expected declaration before '}' token
  113 | }
      | ^
Compiling .pio/build/teensy35/FrameworkArduino/HardwareSerial3.cpp.o
*** [.pio/build/teensy35/liba5a/TeensyStep/Stepper.cpp.o] Error 1
```

</details>

Cheers!

**Comment by luni64** [comment]
Should be fixed in v2.3.3

**Comment by bimac** [comment]
Indeed it is :) thanks for the fix!
